app/api/search.v2/buildQuery.ts

```diff
@@ -1,25 +1,84 @@
 import { SearchQuery } from 'shared/types/SearchQueryType';
 import { RequestBody } from '@elastic/elasticsearch/lib/Transport';
-
 import { cleanUp, searchStringMethod } from './queryHelpers';
 import { permissionsFilters } from './permissionsFilters';
 
-export const buildQuery = async (query: SearchQuery, language: string): Promise<RequestBody> => ({
-  _source: {
-    includes: ['title', 'template', 'sharedId', 'language', 'documents._id'],
-  },
-  query: {
-    bool: {
-      filter: [{ term: { language } }, ...permissionsFilters(query)],
-      must: [
-        query.filter?.searchString && {
-          [await searchStringMethod(query.filter.searchString)]: {
-            query: query.filter.searchString,
+async function extractSearchParams(
+  query: SearchQuery
+): Promise<{
+  searchString?: string | number | undefined;
+  fullTextSearchString?: string;
+  searchMethod: string;
+}> {
+  if (query.filter && query.filter.searchString && typeof query.filter.searchString === 'string') {
+    const { searchString } = query.filter;
+    const searchTypeKey = searchString.includes(':') ? 'searchString' : 'fullTextSearchString';
+    const searchMethod = await searchStringMethod(searchString);
+    return { [searchTypeKey]: searchString, searchMethod };
+  }
+  return {
+    searchString: query.filter?.searchString,
+    searchMethod: 'query_string',
+  };
+}
+
+export const buildQuery = async (query: SearchQuery, language: string): Promise<RequestBody> => {
+  const { searchString, fullTextSearchString, searchMethod } = await extractSearchParams(query);
+  return {
+    _source: {
+      includes: ['title', 'template', 'sharedId', 'language', 'documents._id'],
+    },
+    query: {
+      bool: {
+        filter: [
+          query.filter?.sharedId && {
+            terms: {
+              'sharedId.raw': [query.filter.sharedId],
+            },
+          },
+          {
+            term: { language },
+          },
+          ...permissionsFilters(query),
+        ].filter(cleanUp),
+        must: [
+          fullTextSearchString && {
+            has_child: {
+              type: 'fullText',
+              score_mode: 'max',
+              inner_hits: {
+                _source: false,
+                highlight: {
+                  order: 'score',
+                  pre_tags: ['<b>'],
+                  post_tags: ['</b>'],
+                  encoder: 'html',
+                  number_of_fragments: 9999,
+                  type: 'fvh',
+                  fragment_size: 300,
+                  fragmenter: 'span',
+                  fields: {
+                    'fullText_*': {},
+                  },
+                },
```

Comment:
@mfacar this should only be needed when asking for snippets right ?

Reply:
I extracted the property into a function in order to make it conditional